extensions/bouncer.md

```diff
@@ -0,0 +1,236 @@
+
+The bouncer MAY reject this new network for any reason and MUST then reply with an error message as shown below. If accepted, the bouncer MUST generate a new unique ID for this new network under the users account. The bouncer MAY also include default values for any tags if any are not provided.
+
+The bouncer MAY send the `LISTNETWORKS` reply to all the users connected clients to keep them in sync.
+
+Replies:
+* Network addeed:
+  - `[s] BOUNCER addnetwork netid freenode RPL_OK`
+* Various errors:
+  - `[s] BOUNCER addnetwork * * ERR_NEEDSNAME`
+  - `[s] BOUNCER addnetwork * freenode ERR_NAMEINUSE`
+  - `[s] BOUNCER addnetwork * freenode ERR_INVALIDPORT`
+  - `[s] BOUNCER addnetwork * freenode ERR_MAXNETWORKS`
+  - `[s] BOUNCER addnetwork * freenode ERR_UNKNOWN :Optional extra information`
+
+
+### Changing a network
+Syntax: `[c] BOUNCER changenetwork <netid | *> host=irc.freenode.net;port=6667;nick=prawnsalad;user=prawn`
+
+Change an existing network on the bouncer. The tags are the same as mentioned in the `addnetwork` command. At least one tag MUST be given.
+
+The bouncer MAY send the `LISTNETWORKS` reply to all the users connected clients to keep them in sync.
+
+Replies:
+* Network changed:
+  - `[s] BOUNCER changenetwork netid RPL_OK`
+* Various errors:
+  - `[s] BOUNCER changenetwork * ERR_INVALIDARGS`
+  - `[s] BOUNCER changenetwork netid ERR_NETNOTFOUND`
+  - `[s] BOUNCER changenetwork netid ERR_INVALIDPORT`
+  - `[s] BOUNCER changenetwork netid ERR_UNKNOWN :Optional extra information`
+    
+
+
+### Removing a network
+Syntax: `[c] BOUNCER delnetwork <netid | *>`
+
+Delete a network from the bouncer. If deleting the active network the implimentation may choose how to handle the client connection. Examples include closing it or parting all channels and resorting it to a temporary connection.
+
+The bouncer MAY send the `LISTNETWORKS` reply to all the users connected clients to keep them in sync.
+
+Replies:
+* Network removed:
+  - `[s] BOUNCER delnetwork netid RPL_OK`
+* Various errors:
+  - `[s] BOUNCER delnetwork * ERR_INVALIDARGS`
+  - `[s] BOUNCER delnetwork netid ERR_NETNOTFOUND`
+  - `[s] BOUNCER delnetwork netid ERR_UNKNOWN :Optional extra information`
+
+
+## Network state notifications
+As a network state changes on the server such as connecting or disconnection, it MUST send the changes as they happen to any users connected clients.
+
+* Connection state changes:
+  - `[s] BOUNCER state netid netname connecting`
+  - `[s] BOUNCER state netid netname connected`
+  - `[s] BOUNCER state netid netname disconnected`
+
+## Network and buffer tags
+This document has described several tags that SHOULD be supported. Implementations MAY introduce other tags but MUST contain a vendor prefix, eg. `kiwibnc/mytag=value`.
+
+#### Network tags
+Implementations MUST recognise the following:
+- `network` - The human readable name for the network. It may be changed.
```

Comment:
This reads as "the network attribute of the network". Something like "name" would be better.